printer/printast_with_mappings.ml

```diff
@@ -151,6 +151,16 @@ let arg_label i ppf = function
 let typevars ppf vs =
   List.iter (fun x -> fprintf ppf " %a" Pprintast.tyvar x.txt) vs
 
+let modalities i ppf modes =
+  list i string_loc ppf (
+    List.map (Location.map (fun (Modality x) -> x)) modes
+  )
+
+let modes i ppf modes =
+  list i string_loc ppf (
+    List.map (Location.map (fun (Mode x) -> x)) modes
+  )
+
```

Comment:
I think we should imitate `attributes`. So for each mode, we print `mode` followed by the mode name.